lib/components/drawer/end_drawer.dart

```diff
@@ -0,0 +1,137 @@
+import 'dart:convert';
+import 'dart:ui';
+
+import 'package:flutter/material.dart';
+import 'package:http/http.dart' as http;
+import 'package:provider/provider.dart';
+import 'package:rtchat/components/drawer/sliver_title.dart';
+import 'package:rtchat/models/channels.dart';
+
+class LeftDrawerWidget extends StatefulWidget {
+  const LeftDrawerWidget({Key? key}) : super(key: key);
+
+  @override
+  State<LeftDrawerWidget> createState() => LeftDrawerWidgetState();
+}
+
+class LeftDrawerWidgetState extends State<LeftDrawerWidget> {
+  Future<dynamic>? viewersFuture;
+
+  _viewersFuture(String channel) async {
+    Uri uri = Uri.parse('https://tmi.twitch.tv/group/user/$channel/chatters');
+    final res = await http.get(uri);
+    if (res.statusCode == 200) {
```

Comment:
Use guard clauses instead, like put the error cases in the conditional. In other words,

```
if (res.statusCode != 200) {
  return {...}
}
...decode...
```

As a general rule, the last return should be the "success" case of the function.